src/main/java/io/debezium/connector/cassandra/SchemaHolder.java

```diff
@@ -31,49 +37,64 @@
  * by {@link SchemaProcessor} periodically.
  */
 public class SchemaHolder {
-
     private static final String NAMESPACE = "io.debezium.connector.cassandra";
     private static final Logger LOGGER = LoggerFactory.getLogger(SchemaHolder.class);
 
-    private final Map<KeyspaceTable, KeyValueSchema> tableToKVSchemaMap = new ConcurrentHashMap<>();
+    public final Map<KeyspaceTable, KeyValueSchema> tableToKVSchemaMap = new ConcurrentHashMap<>();
+    public final Multimap<String, String> keyspaceTableMap = Multimaps.synchronizedListMultimap(ArrayListMultimap.create());
 
+    public final String kafkaTopicPrefix;
+    public final SourceInfoStructMaker sourceInfoStructMaker;
     private final CassandraClient cassandraClient;
-    private final String kafkaTopicPrefix;
-    private final SourceInfoStructMaker sourceInfoStructMaker;
 
     public SchemaHolder(CassandraClient cassandraClient, String kafkaTopicPrefix, SourceInfoStructMaker sourceInfoStructMaker) {
         this.cassandraClient = cassandraClient;
         this.kafkaTopicPrefix = kafkaTopicPrefix;
         this.sourceInfoStructMaker = sourceInfoStructMaker;
-        refreshSchemas();
-    }
-
-    public void refreshSchemas() {
-        LOGGER.info("Refreshing schemas...");
-        Map<KeyspaceTable, TableMetadata> latest = getLatestTableMetadatas();
-        removeDeletedTableSchemas(latest);
-        createOrUpdateNewTableSchemas(latest);
-        LOGGER.info("Schemas are refreshed");
     }
 
-    public KeyValueSchema getOrUpdateKeyValueSchema(KeyspaceTable kt) {
-        if (!tableToKVSchemaMap.containsKey(kt)) {
-            refreshSchema(kt);
-        }
+    public synchronized KeyValueSchema getKeyValueSchema(KeyspaceTable kt) {
         return tableToKVSchemaMap.getOrDefault(kt, null);
     }
 
     public Set<TableMetadata> getCdcEnabledTableMetadataSet() {
-        return tableToKVSchemaMap.values().stream()
-                .map(KeyValueSchema::tableMetadata)
-                .filter(tm -> tm.getOptions().isCDC())
+        return cassandraClient
+                .getCluster()
+                .getMetadata()
+                .getKeyspaces()
+                .stream()
+                .map(KeyspaceMetadata::getTables)
+                .flatMap(Collection::stream)
+                .filter(t -> t.getOptions().isCDC())
                 .collect(Collectors.toSet());
     }
 
+    public synchronized void removeKeyspace(String keyspace) {
+        final List<KeyspaceTable> collect = tableToKVSchemaMap.keySet()
+                .stream()
+                .filter(keyValueSchema -> keyValueSchema.keyspace.equals(keyspace))
+                .collect(toList());
+
+        collect.forEach(tableToKVSchemaMap::remove);
+        collect.forEach(collected -> keyspaceTableMap.removeAll(collected.keyspace));
+    }
+
+    public synchronized void removeTableSchema(KeyspaceTable kst) {
+        tableToKVSchemaMap.remove(kst);
```

Comment:
Can you check if `kst` exists in map before trying to remove it please?

Reply:
it is not necessary, IDEA even highlights this

Reply:
> it is not necessary, IDEA even highlights this

It might miss something but I still don't understand why this is not necessary. I'm confused about what `IDEA even highlights this` means. Can you explain more clearly?